visions/using-c++-from-swift.md

```diff
@@ -153,69 +169,82 @@ Or this API which fills in a pointer that has two levels of indirection:
+C++ and Swift use very different models for generic programming. C++ templates are eagerly instantiated for each set of template arguments they're used with, with type checking done separately for each instantiation based on the exact types in use. In contrast, Swift generics are type-checked once based on the requirements they impose on their type parameters, and while they can be specialized for a particular set of type arguments, that is not required or even always possible.
+
+This difference makes using generic C++ APIs in Swift difficult. Generic code in Swift will not be able to use C++ templates generically without substantial new language features and a lot of implementation work. Allowing C++ templates to be used on concrete Swift types is theoretically more feasible but still a major project because of the *ad hoc* nature of type constraints in templates. If this feature is ever pursued, it may require substantial user guidance, such as the thorough adoption of [C++20 concepts constraints](https://en.cppreference.com/w/cpp/language/constraints) on any constrained template used from Swift.
 
-Swift protocols allow C++ types to be used in a generic context in Swift. Users can extend concrete C++ types, or concrete specializations of C++ class templates to conform to a protocol. Swift could provide a tool (likely in the form of an annotation) that allows users to even conform un-specialized templates to Swift protocols. This level of generic programming may be sufficent for most C++ interop users and would allow the Swift compiler to side-step one of the most difficult and complicated parts of C++ interoperability. 
+Fortunately, these limitations do not apply when using C++ types with Swift generics. Unconstrained generics can be used with C++ types without any further work, and programmers can simply add protocol conformances to concrete C++ types in order to use them with constrained generics. Swift could even provide a tool (likely in the form of an annotation) that allows users to give conformances for all specializations of a template. This level of generic programming may be sufficient for most C++ interop users, and it would allow the Swift compiler to side-step one of the most difficult and complicated parts of C++ interoperability.
```

Comment:
> Swift could even provide a tool (likely in the form of an annotation) that allows users to give conformances for all specializations of a template.

We already have this tool! It's the `CONFORMS_TO` attribute. The thing I'm talking about here is importing a `std.vector` itself (the unspecialized class template) as a protocol. 

I'm just going to pull back some of my wording. This topic can be (and is) explored in more depth elsewhere.